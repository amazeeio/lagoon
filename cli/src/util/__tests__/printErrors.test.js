// @flow

import {
  printErrors,
  printNoConfigError,
  printProjectConfigurationError,
  printGraphQLErrors,
} from '../printErrors';

describe('printErrors', () => {
  it('should print objects with a message key', () => {
    const cerr = jest.fn();
    const code = printErrors(cerr, { message: 'arbitraryValue' });
    expect(code).toBe(1);
    expect(cerr.mock.calls).toMatchSnapshot();
  });
});

describe('printNoConfigError', () => {
  it('should print an error about the configuration file not being found', () => {
    const cerr = jest.fn();
    const code = printNoConfigError(cerr);
    expect(code).toBe(1);
    expect(cerr.mock.calls).toMatchSnapshot();
  });
});

describe('printProjectConfigurationError', () => {
  it('should print an error about missing project configuration', () => {
    const cerr = jest.fn();
    const code = printProjectConfigurationError(cerr);
    expect(code).toBe(1);
    expect(cerr.mock.calls).toMatchSnapshot();
  });
});

describe('printGraphQLErrors', () => {
  it('should print singular GraphQLErrors', () => {
    const cerr = jest.fn();
    const code = printGraphQLErrors(cerr, {
      message: 'A GraphQLError message',
    });
    expect(code).toBe(1);
    expect(cerr.mock.calls).toMatchSnapshot();
  });

  it('should print multiple GraphQLErrors', () => {
    const cerr = jest.fn();
    const code = printGraphQLErrors(
      cerr,
      { message: 'A GraphQLError message' },
      { message: 'Another GraphQLError message' },
    );
    expect(code).toBe(1);
    expect(cerr.mock.calls).toMatchSnapshot();
  });

  it('should tell the user to log in if there is a bearer token error', () => {
    const cerr = jest.fn();
    const code = printGraphQLErrors(cerr, {
      message: 'Unauthorized - Bearer Token Required',
    });
    expect(code).toBe(1);
    expect(cerr.mock.calls).toMatchSnapshot();
  });
});
