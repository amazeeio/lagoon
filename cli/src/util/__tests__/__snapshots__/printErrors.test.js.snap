// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`printErrors should print objects with a message key 1`] = `
Array [
  Array [
    "[31marbitraryValue[39m",
  ],
]
`;

exports[`printGraphQLErrors should print multiple GraphQLErrors 1`] = `
Array [
  Array [
    "[31mOops! The Lagoon API returned errors:[39m",
  ],
  Array [
    "[31mA GraphQLError message[39m",
  ],
  Array [
    "[31mAnother GraphQLError message[39m",
  ],
]
`;

exports[`printGraphQLErrors should print singular GraphQLErrors 1`] = `
Array [
  Array [
    "[31mOops! The Lagoon API returned an error:[39m",
  ],
  Array [
    "[31mA GraphQLError message[39m",
  ],
]
`;

exports[`printGraphQLErrors should tell the user to log in if there is a bearer token error 1`] = `
Array [
  Array [
    "[31mNo authentication token found. Please log in first with \\"lagoon login\\".[39m
[31mOnline documentation: https://github.com/amazeeio/lagoon/blob/master/cli/README.md#lagoon-login[39m",
  ],
]
`;

exports[`printNoConfigError should print an error about the configuration file not being found 1`] = `
Array [
  Array [
    "[31m.lagoon.yml config file not found. Please create one with \\"lagoon init\\".[39m
[31mOnline documentation: https://github.com/amazeeio/lagoon/blob/master/cli/README.md#lagoon-init[39m",
  ],
]
`;

exports[`printProjectConfigurationError should print an error about missing project configuration 1`] = `
Array [
  Array [
    "[31mNo project configured. Please create a .lagoon.yml config file with \\"lagoon init\\" or pass a project to this command via the --project option.[39m
[31mOnline documentation: https://github.com/amazeeio/lagoon/blob/master/cli/README.md#lagoon-init[39m",
  ],
]
`;
